xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -107,3 +114,104 @@ async def send_reliable_submission(
     return await _wait_for_final_transaction_outcome(
         transaction_hash, client, prelim_result, transaction.last_ledger_sequence
     )
+
+
+def _decode_tx_blob(tx_blob: str) -> Transaction:
+    """
+    Decodes a transaction blob.
+
+    Args:
+        tx_blob: the tx blob to decode.
+
+    Returns:
+        The formatted transaction.
+    """
+    return Transaction.from_xrpl(decode(tx_blob))
+
+
+async def _get_signed_tx(
+    transaction: Union[Transaction, str],
+    client: Client,
+    wallet: Optional[Wallet] = None,
+    check_fee: bool = True,
+    autofill: bool = True,
+) -> Transaction:
+    """
+    Initializes a signed transaction for a submit request.
+
+    Args:
+        transaction: the transaction or transaction blob to be signed if unsigned.
+        client: the network client to autofill from.
+        wallet: the wallet with which to sign the transaction (optional, only needed
+        if the transaction is not signed).
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+
+    Returns:
+        The signed transaction.
+    """
+    if isinstance(transaction, str):
+        transaction = _decode_tx_blob(transaction)
+
+    if transaction.is_signed():
+        return transaction
+
+    if not wallet:
+        raise XRPLException(
+            "Wallet must be provided when submitting an unsigned transaction"
+        )
+
+    if check_fee:
+        await _check_fee(transaction, client)
+
+    if autofill:
+        transaction = await _autofill(transaction, client)
+
+    return (
+        await sign(transaction, wallet, True)
+        if transaction.signers
+        else await sign(transaction, wallet, True)
```

Comment:
Both sides of this if-statement are the same, which I'm guessing is a bug. I think you meant to do this:
```suggestion
        await sign(transaction, wallet, True, True)
        if transaction.signers
        else await sign(transaction, wallet, True, False)
```

Reply:
Honestly, this will be easier to debug/read if it's a normal if statement instead of trying to put it all in one line